website/addons/owncloud/views.py

```diff
@@ -0,0 +1,130 @@
+"""Views for the node settings page."""
+# -*- coding: utf-8 -*-
+import httplib as http
+
+from flask import request
+from modularodm import Q
+from modularodm.storage.base import KeyExistsException
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.addons.base import generic_views
+from owncloud import Client as OwnCloudClient
+from website.addons.owncloud.model import OwnCloudProvider
+from website.addons.owncloud.settings import DEFAULT_HOSTS
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.oauth.models import ExternalAccount
+from website.project.decorators import (
+    must_have_addon)
+from website.util import api_url_for
+
+from website.addons.owncloud import settings
+
+SHORT_NAME = 'owncloud'
+FULL_NAME = 'OwnCloud'
+
+owncloud_account_list = generic_views.account_list(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_import_auth = generic_views.import_auth(
+    SHORT_NAME,
+    OwnCloudSerializer
+)
+
+owncloud_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+owncloud_root_folder = generic_views.root_folder(
+    SHORT_NAME
+)
+
+@must_be_logged_in
+def owncloud_user_config_get(auth, **kwargs):
+    """View for getting a JSON representation of the logged-in user's
+    OwnCloud user settings.
+    """
+
+    user_addon = auth.user.get_addon('owncloud')
+    user_has_auth = False
+    if user_addon:
+        user_has_auth = user_addon.has_auth
+
+    return {
+        'result': {
+            'userHasAuth': user_has_auth,
+            'urls': {
+                'create': api_url_for('owncloud_add_user_account'),
+                'accounts': api_url_for('owncloud_account_list'),
+            },
+            'hosts': DEFAULT_HOSTS,
+        },
+    }, http.OK
+
+
+## Config ##
+
+@must_be_logged_in
+def owncloud_add_user_account(auth, **kwargs):
+    """Verifies new external account credentials and adds to user's list"""
+    user = auth.user
+
+    host = request.json.get('host').rstrip('/')
+    if 'https://' not in host:
+        host = 'https://' + host
+    username = request.json.get('username')
+    password = request.json.get('password')
+    provider = OwnCloudProvider(account=None, host=host, username=username, password=password)
+
+    try:
+        oc = OwnCloudClient(host, verify_certs=settings.USE_SSL)
+        oc.login(username, password)
+        oc.logout()
+    except:
+        return {
+            'message': 'All the fields above are required.'
```

Comment:
This message doesn't seem to fit with what could be happening here. That is, supplying all of the required fields but with any one of them incorrect would lead to a `400` telling you that you need to supply all the required fields. This check *should* be done, but probably above rather than here.